examples/ecs/state.rs

```diff
@@ -6,26 +6,31 @@ use bevy::prelude::*;
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
-        .add_state(AppState::Menu)
+        .add_state::<AppState>()
         .add_startup_system(setup)
-        .add_system_set(SystemSet::on_enter(AppState::Menu).with_system(setup_menu))
-        .add_system_set(SystemSet::on_update(AppState::Menu).with_system(menu))
-        .add_system_set(SystemSet::on_exit(AppState::Menu).with_system(cleanup_menu))
-        .add_system_set(SystemSet::on_enter(AppState::InGame).with_system(setup_game))
-        .add_system_set(
-            SystemSet::on_update(AppState::InGame)
-                .with_system(movement)
-                .with_system(change_color),
-        )
+        .add_system_to_schedule(setup_menu, &OnEnter(AppState::Menu))
+        .add_system(menu.on_update(AppState::Menu))
+        .add_system_to_schedule(cleanup_menu, &OnExit(AppState::Menu))
+        .add_system_to_schedule(setup_game, &OnEnter(AppState::InGame))
+        .add_systems((movement, change_color).on_update(AppState::InGame))
```

Comment:
Will this eventually become as below?
```rust
        .add_system(setup_menu.on_enter(AppState::Menu))
        .add_system(menu.on_update(AppState::Menu))
        .add_system(cleanup_menu.on_exit(AppState::Menu))
        .add_system(setup_game.on_enter(AppState::InGame))
        .add_systems((movement, change_color).on_update(AppState::InGame))
```

I find the `add_system_to_schedule` API a bit confusing. It also seems to be used only in userland, while Bevy internally doesn't have to.

I have good experience with Bevy but I haven't followed the Stageless discussions much. The way today I would read the above is:

"Ok so I'm adding `setup_menu` to the schedule that is run when entering the `Menu` state, then I'm _not_ adding the `menu` system to a schedule but it does run when updating the `Menu` state, why?" etc.

It took some digging to find out about `CoreSchedule::Main`.

From what I see it seems to be used for `on_enter` and `on_exit` and for `CoreSchedule::FixedTimestep`, if so, maybe it could be removed in favour of  the syntax `.on_enter(State)` / `.on_exit(State)` syntax and add `.add_fixed_timestep_system` (like we have `.add_startup_system` being automatically added to `CoreSchedule::Startup`)?

If it's necessary to expose `add_system_to_schedule`, then in my opinion it should be an advanced edge case and not be shown in beginner facing examples, if possible.

(PS: it's very likely this has already been picked up / discussed, just commenting on what I see here).

Reply:
> if so, maybe it could be removed in favour of the syntax .on_enter(State) / .on_exit(State) syntax and add .add_fixed_timestep_system

The latter is trivial, the former would require yet another layer of traits, as it only makes sense at the "multiple schedules" level. I think I'm in favor, but I'd like to defer to a later PR.